piker/ui/_search.py

```diff
@@ -153,23 +158,55 @@ def set_font_size(self, size: int = 18):
 
         self.setStyleSheet(f"font: {size}px")
 
-    def resize(self):
+    # def resizeEvent(self, event: 'QEvent') -> None:
+    #     self.resize_to_results()
+    #     super().resizeEvent(event)
+
+    def on_resize(self) -> None:
+        '''
+        Resize relay event from god.
+
+        '''
+        self.resize_to_results()
+
+    def resize_to_results(self):
         model = self.model()
         cols = model.columnCount()
 
+        col_w_tot = 0
         for i in range(cols):
             self.resizeColumnToContents(i)
+            col_w_tot += self.columnWidth(i)
 
-        # inclusive of search bar and header "rows" in pixel terms
-        rows = 100
-        # max_rows = 8  # 6 + search and headers
         row_px = self.rowHeight(self.currentIndex())
-        # print(f'font_h: {font_h}\n px_height: {px_height}')
 
         # TODO: probably make this more general / less hacky
-        self.setMinimumSize(self.width(), rows * row_px)
-        self.setMaximumSize(self.width() + 10, rows * row_px)
-        self.setFixedWidth(333)
+        # we should figure out the exact number of rows to allow
+        # inclusive of search bar and header "rows", in pixel terms.
+        if row_px > 0:
+            window_h = self.window().height()
+            rows = ceil(window_h / row_px) - 2
+        else:
+            rows = 16
+
+        # TODO: the problem here is that this view widget is **not**
+        # resizing/scaling when the parent layout is adjusted, not sure
+        # what exactly is up... only "scale up" the results view
+        # when the window size has increased/
+        if not self._last_window_h or self._last_window_h < window_h:
+            self.setMaximumSize(self.width(), rows * row_px)
+            self.setMinimumSize(self.width(), rows * row_px)
+
+        # self.resize(self.width(), rows * row_px)
+        self.resize(self.width(), rows * row_px)
+        self._last_window_h = window_h
+
+        # size to width of longest result seen thus far
+        # TODO: should we always dynamically scale to longest result?
+        if self.width() < col_w_tot:
```

Comment:
As per my comment, we should just always size the largest result it just makes the UX a little *jittery* when results are filling out 😂 

I'm not sure which i like better.
I guess we could also add a bit of state tracking between each `.show()` of the search such that it only grows from the last "session" (aka since showing the widget) of results?

I dunno. give feeback!